src/router/validation/jsonSchemaValidator.ts

```diff
@@ -16,15 +16,24 @@ import fhirV3Schema from './schemas/fhir.schema.v3.json';
 export default class JsonSchemaValidator implements Validator {
     private ajv: any;
 
+    private readonly schemaId: string;
+
     constructor(fhirVersion: FhirVersion) {
         const ajv = new Ajv({ schemaId: 'auto', allErrors: true });
+        let schema;
         if (fhirVersion === '4.0.1') {
             ajv.addMetaSchema(schemaDraft06);
             ajv.addSchema(fhirV4Schema);
-        }
-        if (fhirVersion === '3.0.1') {
+            schema = fhirV4Schema;
+        } else if (fhirVersion === '3.0.1') {
             ajv.addMetaSchema(schemaDraft04);
             ajv.addSchema(fhirV3Schema);
+            schema = fhirV3Schema;
+        }
+        if (schema && 'id' in schema) {
+            this.schemaId = schema.id;
+        } else {
+            this.schemaId = ``;
         }
```

Comment:
nit: Can simplify to single line ternary
```suggestion
        this.schemaId = (schema && schema.id) ? schema.id : '';
```

Reply:
I changed that statement into a ternary. The above didn't work for me because we need to also incorporate `id in schema` condition.